File: client/src/utils/language.ts
/**
 * Language detection utilities.
 */

// Hebrew Unicode range: \u0590-\u05FF
const HEBREW_REGEX = /[\u0590-\u05FF]/;

/**
 * Check if text contains Hebrew characters.
 */
export function isHebrew(text: string): boolean {
  return HEBREW_REGEX.test(text);
}

/**
 * Get text direction based on content.
 */
export function getTextDirection(text: string): 'rtl' | 'ltr' {
  return isHebrew(text) ? 'rtl' : 'ltr';
}

/**
 * Get direction for a language code.
 */
export function getLanguageDirection(language: string): 'rtl' | 'ltr' {
  return language === 'he' ? 'rtl' : 'ltr';
}

export type Language = 'en' | 'he';

File: server/app/utils/language.py
"""Language detection utilities."""

import re

# Hebrew Unicode range: \u0590-\u05FF (Hebrew letters and marks)
HEBREW_PATTERN = re.compile(r"[\u0590-\u05FF]")


def detect_language(text: str) -> str:
    """Detect language from text content.

    Returns 'he' if Hebrew characters are found, otherwise 'en'.
    """
    if HEBREW_PATTERN.search(text):
        return "he"
    return "en"


def is_hebrew(text: str) -> bool:
    """Check if text contains Hebrew characters."""
    return bool(HEBREW_PATTERN.search(text))
